graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/AggregationTabularResponseCreator.java

```diff
@@ -40,31 +41,43 @@ public class AggregationTabularResponseCreator implements TabularResponseCreator
 
     public TabularResponse mapToResponse(final AggregationRequestSpec searchRequestSpec, final SearchJob searchJob) throws AggregationFailedException {
         final SearchJobDTO searchJobDTO = SearchJobDTO.fromSearchJob(searchJob);
+
+
         final QueryResult queryResult = searchJobDTO.results().get(SearchRequestSpecToSearchMapper.QUERY_ID);
 
         if (queryResult != null) {
             throwErrorIfAnyAvailable(queryResult);
             final SearchType.Result aggregationResult = queryResult.searchTypes().get(AggregationSpecToPivotMapper.PIVOT_ID);
+
             if (aggregationResult instanceof PivotResult pivotResult) {
-                return mapToResponse(searchRequestSpec, pivotResult);
+                final List<SeriesSpec> requestedSeries = extractSeries(queryResult);
+                return mapToResponse(searchRequestSpec, pivotResult, requestedSeries);
             }
         }
 
         LOG.warn("Scripting API failed to obtain aggregation for input : " + searchRequestSpec);
         throw new AggregationFailedException("Scripting API failed to obtain aggregation for input : " + searchRequestSpec);
     }
 
+    private List<SeriesSpec> extractSeries(QueryResult queryResult) {
```

Comment:
I'd rename it to "extractSeriesSpec". I'd consider replacing "series" with "seriesSpec" in other places as well, to clearly state that we do not have any data there, it is some kind of specification that allows us to fetch those data.

Reply:
good idea, done